lib/src/pubspec.dart

```diff
@@ -635,3 +690,63 @@ class Pubspec extends PubspecBase {
 /// This is slightly more complicated than just checking if the scheme is
 /// 'file', since relative URIs also refer to the filesystem on the VM.
 bool _isFileUri(Uri uri) => uri.scheme == 'file' || uri.scheme == '';
+
+/// Merges [overrides] from an overrides file into a [pubspec] file.
+///
+/// The merging is performed at the stage where the pubspec and overrides files
+/// have both been parsed into [YamlMap]s. Source references of [YamlNode]s are
+/// preserved, so that error messages will point to the correct source location.
+///
+/// Only the the following top-level fields of a pubspec are merged:
+///
+/// - `dependencies`
+/// - `dev_dependencies`
```

Comment:
What is the value in overriding `dependencies` and `dev_dependencies`, when will that work?

Reply:
If you override `dependencies` to use a path dependency, then anything with a transitive dependency on the path-dependency will have a problem.. because the source of a dependency is part of its identity.

So if you override `dependencies` then you most likely end up using `dependency_overrides`.

Reply:
You're right, we should only support `dependency_overrides` in `pubspec_overrides.yaml`.

Reply:
Done